src/Levels/base_level.py

```diff
@@ -38,8 +41,26 @@ def update(self):
             self.keep_running = False
 
         self.player_group.update(self.user_interface.movePlayerCommand)
-        self.player_group.sprite.attack = self.user_interface.attackCommand
         self.visible_ghosts_group.update()
+        if self.user_interface.attackCommand:
+            self.do_attack(self._player)
+
+    def do_attack(self, player):
+        """
+        Check whether we are in the zone of application of user's weapon.
+        If so, decrease the number of ghosts.
+        """
+        for qghost in self.ghosts_group:
+            for i, ghost in enumerate(qghost.visible_parts):
+                if is_ghost_in_players_radius(player.position, ghost.position):
+                    # TODO: so far it's classical. Replace by quantum
+                    if qghost.quantum_state[i] == 1:
+                        qghost.quantum_state.pop(i)
+                        self.visible_ghosts_group.remove(qghost.visible_parts.pop(i))
+                    else:
+                        qghost.quantum_state[i] -= 1
+            if not qghost.visible_parts:
+                self.ghosts_group.remove(qghost)
```

Comment:
I think this belongs to the `Player` class, because it is an action of the player. When doing the `update` function of the player, make also a call to `do_attack`. You can pass the ghosts as parameters to the player's `update` function by passing to the `player_group.update`

Reply:
Makes sense!